react/SliderLayout.tsx

```diff
@@ -1,14 +1,88 @@
-import React, { FC } from 'react'
-import { SliderContextProvider } from './components/SliderContext'
+import React from 'react'
+import { defineMessages } from 'react-intl'
+
 import Slider from './components/Slider'
+import { SliderContextProvider } from './components/SliderContext'
+
+const SliderLayout: StorefrontFunctionComponent<
+  SliderLayoutProps & SliderLayoutSiteEditorProps
+> = ({
+  totalItems,
+  infinite = false,
+  showNavigationArrows = 'always',
+  showPaginationDots = 'always',
+  usePagination = true,
+  children,
+  ...contextProps
+}) => {
+  const totalSlides = totalItems || React.Children.count(children)
 
-const SliderLayout: FC<SliderLayoutProps> = props => {
-  const totalItems = React.Children.count(props.children)
   return (
-    <SliderContextProvider totalItems={totalItems} {...props}>
-      <Slider>{props.children}</Slider>
+    <SliderContextProvider totalItems={totalSlides} {...contextProps}>
+      <Slider
+        infinite={infinite}
+        showNavigationArrows={showNavigationArrows}
+        showPaginationDots={showPaginationDots}
+        totalItems={totalSlides}
+        usePagination={usePagination}
+      >
+        {children}
+      </Slider>
     </SliderContextProvider>
   )
 }
 
+const messages = defineMessages({
+  sliderTitle: {
+    id: 'admin/editor.slider-layout.title',
+    defaultMessage: '',
+  },
+  sliderInfinite: {
+    id: 'admin/editor.slider-layout.infinite',
+    defaultMessage: '',
+  },
+  sliderShowNavigation: {
+    id: 'admin/editor.slider-layout.showNavigation',
+    defaultMessage: '',
+  },
+  sliderShowPaginationDots: {
+    id: 'admin/editor.slider-layout.showPaginationDots',
+    defaultMessage: '',
+  },
+  sliderUsePagination: {
+    id: 'admin/editor.slider-layout.usePagination',
+    defaultMessage: '',
+  },
+})
+
+SliderLayout.schema = {
+  title: messages.sliderTitle.id,
+  description: messages.sliderTitle.id,
+  type: 'object',
+  properties: {
+    infinite: {
+      default: true,
+      title: messages.sliderInfinite,
+      type: 'boolean',
+    },
+    showNavigationArrows: {
+      default: 'always',
+      enum: ['mobileOnly', 'desktopOnly', 'always', 'never'],
+      title: messages.sliderShowNavigation,
+      type: 'string',
+    },
+    showPaginationDots: {
+      default: 'always',
+      enum: ['mobileOnly', 'desktopOnly', 'always', 'never'],
```

Comment:
A const would be more appropriate, since you need to use in many places

Reply:
This array of values is only being used inside this object `SliderLayout.schema`, and just twice. I think it doesn't hurt but actually improves readability if the enum definitions are written like that. If this eventually gets used somewhere else I'll save it in a variable :)